pyart/io/nexrad_level2.py

```diff
@@ -584,14 +587,30 @@ def _get_record_from_buf(buf, pos):
     if msg_type == 31:
         new_pos = _get_msg31_from_buf(buf, pos, dic)
     elif msg_type == 5:
-        new_pos = _get_msg5_from_buf(buf, pos, dic)
+        # Sometimes we encounter incomplete buffers
+        try:
+            new_pos = _get_msg5_from_buf(buf, pos, dic)
+        except struct.error:
+            warnings.warn("Encountered incomplete MSG5. File may be corrupt.",
+                          RuntimeWarning)
+            new_pos = pos + RECORD_SIZE
+    elif msg_type == 29:
+        new_pos = _get_msg29_from_buf(buf, pos, dic)
+        warnings.warn("Message 29 encountered, not parsing.", RuntimeWarning)
     elif msg_type == 1:
-        new_pos = _get_msg1_from_buf(buf, pos, dic)
+        new_pos = _get_msg1_from_buf(buf, pos, dic)       
     else:   # not message 31 or 1, no decoding performed
         new_pos = pos + RECORD_SIZE
 
     return new_pos, dic
 
+def _get_msg29_from_buf(buf, pos, dic):
+    msg_size = dic['header']['size']
+    if msg_size == 65535:
```

Comment:
Just to be clear, while this format change was made to accommodate message type 29, there's nothing in the NEXRAD ICD that indicates that this special sizing only applies to message 29. It's just that 29 is the only data we get currently that seems to use this.